// providers/esm2015/lib/providers/provider-rating/provider-rating.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { HttpClient } from '@angular/common/http';
import { MatSnackBar } from '@angular/material/snack-bar';
import { ProvidersService } from '@brightpath/scheduling';
let ProviderRatingComponent = class ProviderRatingComponent {
    constructor(providerService, http, snackbar, router, route) {
        this.providerService = providerService;
        this.http = http;
        this.snackbar = snackbar;
        this.router = router;
        this.route = route;
        this.providerId = '';
        this.showWidget = false;
    }
    ngOnInit() {
        this.route.params.subscribe((params) => {
            console.log(params);
            if (params.id) {
                this.providerId = params.id;
            }
            this.showWidget = true;
        });
    }
};
ProviderRatingComponent.ctorParameters = () => [
    { type: ProvidersService },
    { type: HttpClient },
    { type: MatSnackBar },
    { type: Router },
    { type: ActivatedRoute }
];
ProviderRatingComponent = __decorate([
    Component({
        selector: 'provider-rating',
        template: "<div class=\"page-layout blank p-24\" fusePerfectScrollbar>\n  <provider-rating-widget *ngIf=\"showWidget\" [providerId]=\"providerId\"></provider-rating-widget>\n</div>\n",
        styles: [""]
    })
], ProviderRatingComponent);
export { ProviderRatingComponent };
