// providers/esm5/lib/providers/provider-widget/provider-widget.component.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Provider } from '@brightpath/common';
import { ProvidersService } from '@brightpath/scheduling';
var ProviderWidgetComponent = /** @class */ (function () {
    function ProviderWidgetComponent(providerService, http, cdr) {
        this.providerService = providerService;
        this.http = http;
        this.cdr = cdr;
        this.provider = new Provider({ id: 'unknown' });
        this.designatedProvider = false;
        this.onAppointmentCreated = new EventEmitter();
        this.onSelectedProvider = new EventEmitter();
        // tslint:disable-next-line:variable-name
        this._showAvailability = false;
        this.availabilityLoaded = false;
        this.showRating = false;
        this.showDetails = false;
        this.showPrivateMessage = false;
        this.loaded = false;
    }
    ProviderWidgetComponent.prototype.selectProvider = function (value) {
        console.log(value);
        if (value.checked) {
            this.onSelectedProvider.emit(this.provider);
        }
    };
    ProviderWidgetComponent.prototype.appointmentCreated = function (appointment) {
        this.onAppointmentCreated.emit(appointment);
    };
    ProviderWidgetComponent.prototype.clearAvailability = function () {
        this.showAvailability = false;
        this.availabilityLoaded = false;
    };
    ProviderWidgetComponent.prototype.onAvailabilityLoaded = function (event) {
        console.log('bam.  loaded');
        this.availabilityLoaded = true;
        this.cdr.detectChanges();
    };
    Object.defineProperty(ProviderWidgetComponent.prototype, "showAvailability", {
        get: function () {
            return this._showAvailability;
        },
        set: function (availability) {
            this._showAvailability = availability;
            this.cdr.detectChanges();
        },
        enumerable: true,
        configurable: true
    });
    ProviderWidgetComponent.prototype.onRatingSaved = function (event) {
        this.showRating = false;
        this.cdr.detectChanges();
    };
    ProviderWidgetComponent.prototype.ngOnInit = function () {
        var _this = this;
        // load the survey results for the provider
        if (this.provider.id === 'unknown') {
            this.providerService.getProviderById(this.providerId)
                .then(function (response) {
                _this.provider = response;
                _this.cdr.detectChanges();
            }, function (error) {
                console.log('An error occurred');
            });
        }
        else {
            this.providerId = this.provider.id;
        }
        var surveyUrl = "https://api-dev.brightpathvt.com/providers/surveyresults/" + this.providerId;
        this.http.get(surveyUrl)
            .subscribe(function (response) {
            _this.loaded = true;
            _this.providerDetails = response;
            console.log('details', response);
            // const licenses: any[] = this.providerDetails.licenses;
            // const licenseString = licenses.map((license: any) => {
            //   console.log('license', license);
            //   return license.name;
            // }).join(',');
            // console.log('licenseString', licenseString);
            _this.cdr.detectChanges();
        }, function (error) {
            _this.loaded = true;
            console.log('An error occurred');
            console.log(error);
        });
    };
    ProviderWidgetComponent.ctorParameters = function () { return [
        { type: ProvidersService },
        { type: HttpClient },
        { type: ChangeDetectorRef }
    ]; };
    __decorate([
        Input()
    ], ProviderWidgetComponent.prototype, "provider", void 0);
    __decorate([
        Input()
    ], ProviderWidgetComponent.prototype, "providerId", void 0);
    __decorate([
        Input()
    ], ProviderWidgetComponent.prototype, "designatedProvider", void 0);
    __decorate([
        Output()
    ], ProviderWidgetComponent.prototype, "onAppointmentCreated", void 0);
    __decorate([
        Output()
    ], ProviderWidgetComponent.prototype, "onSelectedProvider", void 0);
    ProviderWidgetComponent = __decorate([
        Component({
            selector: 'provider-widget',
            template: "<mat-card fxLayout=\"column\" class=\"p-0\">\n\n  <!-- Front -->\n  <div class=\"mat-white-bg mat-elevation-z2\">\n    <div fxLayout=\"row\" *ngIf=\"!designatedProvider\" class=\"p-16 mb-24 bb\">\n      <div class=\"mr-16\">\n        <mat-checkbox (change)=\"selectProvider($event)\"></mat-checkbox>\n      </div>\n      <div fxLayout=\"column\">\n        <div>Choose {{provider?.firstName}} as your preferred provider</div>\n      </div>\n    </div>\n    <div *ngIf=\"!showAvailability && !showRating && !showDetails && !showPrivateMessage\" class=\"mb-8\" fxLayout=\"column\">\n      <div *ngIf=\"designatedProvider\" class=\"font-size-18 text-center p-16 mb-24 bb\">\n        Your Counselor\n      </div>\n      <div fxLayout=\"column\" fxLayoutAlign=\"center center\">\n        <div>\n          <img *ngIf=\"loaded\" src=\"{{ provider?.imageUrl }}\" style=\"border-radius: 50%; width: 150px;\">\n          <mat-progress-spinner *ngIf=\"!loaded\" [strokeWidth]=\"4\" [diameter]=\"150\" mode=\"indeterminate\" class=\"mb-16\"></mat-progress-spinner>\n        </div>\n        <div *ngIf=\"loaded\" class=\"mt-24 mb-8\">\n          <div class=\"secondary-text font-size-18\">{{ provider?.firstName }} {{ provider?.lastName }}</div>\n        </div>\n        <div *ngIf=\"loaded\" class=\"text-center mb-16\" fxLayoutAlign=\"row\">\n          <mat-icon class=\"yellow-icon\">star</mat-icon>\n          <mat-icon class=\"yellow-icon\">star</mat-icon>\n          <mat-icon class=\"yellow-icon\">star</mat-icon>\n          <mat-icon class=\"yellow-icon\">star</mat-icon>\n          <mat-icon class=\"yellow-icon\">star</mat-icon>\n        </div>\n      </div>\n    </div>\n    <div *ngIf=\"showAvailability\" class=\"pt-8\" fxLayout=\"column\">\n      <div  *ngIf=\"designatedProvider\" class=\"mb-8\" fxLayout=\"row\">\n        <div fxFlex class=\"bb pb-8 pl-16 mat-title\">Your Counselor</div>\n      </div>\n      <div fxLayout=\"column\" class=\"mb-16\">\n        <div fxLayout=\"row\" fxLayoutAlign=\"start center\">\n          <mat-spinner class=\"ml-8 mr-8\" *ngIf=\"!availabilityLoaded\" [diameter]=\"25\"></mat-spinner>\n          <button *ngIf=\"availabilityLoaded\" mat-icon-button (click)=\"clearAvailability()\">\n            <mat-icon aria-label=\"Example icon-button with a heart icon\">arrow_back</mat-icon>\n          </button>\n          <div fxLayoutAlign=\"column\" class=\"mr-8 ml-8\">\n            <img src=\"{{ provider?.imageUrl }}\" style=\"border-radius: 50%; width: 40px; height: 40px;\">\n          </div>\n          <div fxLayout=\"column\">\n            <div class=\"secondary-text font-size-18 line-height-14\">{{ provider?.firstName }}</div>\n            <div class=\"secondary-text font-size-24\">{{ provider?.lastName }}</div>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div *ngIf=\"showRating\" class=\"pt-8\" fxLayout=\"column\">\n      <div *ngIf=\"designatedProvider\" class=\"mb-8\" fxLayout=\"row\">\n        <div fxFlex class=\"bb pb-8 pl-16 mat-title\">Your Counselor</div>\n      </div>\n      <div fxLayout=\"column\" class=\"mb-16\">\n        <div fxLayout=\"row\" fxLayoutAlign=\"start center\">\n          <button mat-icon-button (click)=\"showRating = false;\">\n            <mat-icon aria-label=\"Cancel\">arrow_back</mat-icon>\n          </button>\n          <div fxLayoutAlign=\"column\" class=\"mr-8 ml-8\">\n            <img src=\"{{ provider?.imageUrl }}\" style=\"border-radius: 50%; width: 40px; height: 40px;\">\n          </div>\n          <div fxLayoutAlign=\"row\" class=\"secondary-text font-size-24\">Rate {{ provider?.firstName }}</div>\n        </div>\n      </div>\n    </div>\n    <div *ngIf=\"showDetails\" class=\"pt-8\" fxLayout=\"column\">\n      <div *ngIf=\"designatedProvider\" class=\"mb-8\" fxLayout=\"row\">\n        <div fxFlex class=\"bb pb-8 pl-16 mat-title\">Your Counselor</div>\n      </div>\n      <div fxLayout=\"column\" class=\"mb-16\">\n        <div fxLayout=\"row\" fxLayoutAlign=\"start center\">\n          <button mat-icon-button (click)=\"showDetails = false;\">\n            <mat-icon aria-label=\"Cancel\">arrow_back</mat-icon>\n          </button>\n          <div fxLayoutAlign=\"column\" class=\"mr-8 ml-8\">\n            <img src=\"{{ provider?.imageUrl }}\" style=\"border-radius: 50%; width: 40px; height: 40px;\">\n          </div>\n          <div fxLayoutAlign=\"row\" class=\"secondary-text font-size-24\">Meet {{ provider?.firstName }}</div>\n        </div>\n      </div>\n    </div>\n    <div *ngIf=\"showPrivateMessage\" class=\"pt-8\" fxLayout=\"column\">\n      <div *ngIf=\"designatedProvider\" class=\"mb-8\" fxLayout=\"row\">\n        <div fxFlex class=\"bb pb-8 pl-16 mat-title\">Your Counselor</div>\n      </div>\n      <div fxLayout=\"column\" class=\"mb-16\">\n        <div fxLayout=\"row\" fxLayoutAlign=\"start center\">\n          <button mat-icon-button (click)=\"showPrivateMessage = false;\">\n            <mat-icon aria-label=\"Cancel\">arrow_back</mat-icon>\n          </button>\n          <div fxLayoutAlign=\"column\" class=\"mr-8 ml-8\">\n            <img src=\"{{ provider?.imageUrl }}\" style=\"border-radius: 50%; width: 40px; height: 40px;\">\n          </div>\n          <div fxLayoutAlign=\"row\" class=\"secondary-text font-size-24\">Contact {{ provider?.firstName }}</div>\n        </div>\n      </div>\n    </div>\n    <provider-private-message (close)=\"showPrivateMessage = false\" [providerId]=\"provider.id\" *ngIf=\"showPrivateMessage\"></provider-private-message>\n    <provider-rating-widget (saved)=\"onRatingSaved($event)\" *ngIf=\"showRating\" [provider]=\"provider\"></provider-rating-widget>\n    <schedule-availability (appointmentCreated)=\"appointmentCreated($event)\" (availabilityLoaded)=\"onAvailabilityLoaded($event)\" *ngIf=\"showAvailability\" [providerId]=\"provider.id\"></schedule-availability>\n    <provider-details [providerDetails]=\"providerDetails\" *ngIf=\"showDetails\"></provider-details>\n\n    <div *ngIf=\"loaded\" class=\"py-8 grey-50-bg border-top\" fxLayout=\"column\">\n      <mat-nav-list>\n        <mat-list-item (click)=\"showDetails = true\">\n          <mat-icon matListIcon>person</mat-icon>\n          <div matLine>{{ provider?.firstName }}'s Profile</div>\n        </mat-list-item>\n        <mat-list-item (click)=\"showRating = true\">\n          <mat-icon matListIcon>star_rate</mat-icon>\n          <div matLine>Rate {{ provider?.firstName }}</div>\n        </mat-list-item>\n        <mat-list-item (click)=\"showAvailability = true\" *ngIf=\"provider.stripeAccountCreated == true\">\n          <mat-icon matListIcon>date_range</mat-icon>\n          <div matLine>Schedule an Appointment</div>\n        </mat-list-item>\n        <!--<mat-list-item (click)=\"showAvailability = true\" *ngIf=\"provider.stripeAccountCreated == false\">-->\n          <!--<mat-icon matListIcon>date_range</mat-icon>-->\n          <!--<div matLine>Request an Appointment</div>-->\n        <!--</mat-list-item>-->\n        <mat-list-item (click)=\"showPrivateMessage = true\">\n          <mat-icon matListIcon>message</mat-icon>\n          <div matLine>Send a Private Message</div>\n        </mat-list-item>\n      </mat-nav-list>\n    </div>\n  </div>\n</mat-card>\n",
            styles: [".yellow-icon{color:#fdd835}.yellow-icon svg{fill:#fdd835}"]
        })
    ], ProviderWidgetComponent);
    return ProviderWidgetComponent;
}());
export { ProviderWidgetComponent };
