// providers/esm5/lib/providers/provider-details/provider-details.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { HttpClient } from '@angular/common/http';
var ProviderDetailsComponent = /** @class */ (function () {
    function ProviderDetailsComponent(http) {
        this.http = http;
        this.loading = false;
    }
    ProviderDetailsComponent.prototype.ngOnInit = function () {
    };
    ProviderDetailsComponent.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    __decorate([
        Input()
    ], ProviderDetailsComponent.prototype, "providerDetails", void 0);
    ProviderDetailsComponent = __decorate([
        Component({
            selector: 'provider-details',
            template: "<div class=\"px-8\">\n  <div class=\"secondary-text text-bold font-size-16 mb-8\">\n    Bio\n  </div>\n  <div class=\"px-16 mb-16\" [innerHtml]=\"providerDetails.body.description | keepHtml\">\n  </div>\n  <div class=\"mb-16\">\n    <div class=\"secondary-text text-bold font-size-16 mb-8\">Licenses</div>\n    <div class=\"px-16\" *ngFor=\"let currentLicence of providerDetails.body?.licenses\">\n      <div class=\"secondary-text text-bold\">\n        {{ currentLicence.name }}\n      </div>\n      <div fxLayout=\"row\">\n        <div fxFlex=\"25\">\n          State:\n        </div>\n        <div fxFlex=\"25\">\n          {{ currentLicence.state}}\n        </div>\n        <div fxFlex=\"25\">\n          Expires\n        </div>\n        <div fxFlex=\"25\">\n          {{ currentLicence.expiresYear}}\n        </div>\n      </div>\n    </div>\n  </div>\n  <div class=\"secondary-text text-bold font-size-16 mb-8\">\n    Specialties\n  </div>\n  <div>\n    <ul>\n      <li *ngFor=\"let expertise of providerDetails.body?.expertise\">\n        {{ expertise }}\n      </li>\n    </ul>\n  </div>\n\n</div>\n",
            styles: [""]
        })
    ], ProviderDetailsComponent);
    return ProviderDetailsComponent;
}());
export { ProviderDetailsComponent };
