// providers/esm5/lib/providers/provider-rating/provider-rating-widget/provider-rating-widget.component.js
import { __decorate } from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { HttpClient } from '@angular/common/http';
import { CustomerReview } from '../customer-review';
import { ActivatedRoute, Router } from '@angular/router';
import { MatSnackBar } from '@angular/material/snack-bar';
import { ProvidersService } from '@brightpath/scheduling';
var ProviderRatingWidgetComponent = /** @class */ (function () {
    function ProviderRatingWidgetComponent(providerService, http, snackbar, router, route) {
        this.providerService = providerService;
        this.http = http;
        this.snackbar = snackbar;
        this.router = router;
        this.route = route;
        this.providerId = '';
        this.customerId = '';
        this.rating = 0;
        this.ratings = [];
        this.commentFormControl = new FormControl('', [Validators.required]);
        this.loading = false;
        this.loadingReviews = false;
        this.saved = new EventEmitter();
    }
    ProviderRatingWidgetComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.loadingReviews = true;
        this.providerService.getProviderById(this.provider.id)
            .then(function (response) {
            console.log(response);
            _this.provider = response;
        }, function (error) {
            console.log('Error getting provider');
            console.log(error);
        });
        this.providerService.getRatingsForProvider(this.provider.id)
            .then(function (response) {
            console.log('success', response);
            _this.ratings = response;
            _this.loadingReviews = false;
        });
    };
    ProviderRatingWidgetComponent.prototype.onRatingChanged = function (rating) {
        console.log('Changed: ' + rating);
        this.rating = rating;
    };
    ProviderRatingWidgetComponent.prototype.submitRating = function () {
        var _this = this;
        var customerReview = new CustomerReview();
        customerReview.id = this.provider.id;
        customerReview.range = this.customerId;
        customerReview.submittedOn = new Date().getTime();
        customerReview.rating = this.rating;
        customerReview.comments = this.commentFormControl.value;
        var reviewUrl = "https://api-dev.brightpathvt.com/customers/reviews";
        this.loading = true;
        console.log('customerreview', customerReview);
        this.http.post(reviewUrl, customerReview)
            .subscribe(function (response) {
            _this.loading = false;
            _this.snackbar.open('Review Saved', 'OK', {
                verticalPosition: 'top',
                duration: 2000
            });
            // show the success message
            _this.saved.emit(response);
            // this.router.navigateByUrl('/home');
        }, function (error) {
            _this.loading = false;
        }, function () {
            _this.loading = false;
        });
    };
    ProviderRatingWidgetComponent.ctorParameters = function () { return [
        { type: ProvidersService },
        { type: HttpClient },
        { type: MatSnackBar },
        { type: Router },
        { type: ActivatedRoute }
    ]; };
    __decorate([
        Input()
    ], ProviderRatingWidgetComponent.prototype, "provider", void 0);
    __decorate([
        Input()
    ], ProviderRatingWidgetComponent.prototype, "providerId", void 0);
    __decorate([
        Input()
    ], ProviderRatingWidgetComponent.prototype, "customerId", void 0);
    __decorate([
        Output()
    ], ProviderRatingWidgetComponent.prototype, "saved", void 0);
    ProviderRatingWidgetComponent = __decorate([
        Component({
            selector: 'provider-rating-widget',
            template: "<div fxLayout=\"column\" class=\"p-16 orange-50-bg\">\n  <div>\n    1.  On a scale of five stars, how pleased are you with the counseling services you have received from {{ provider?.firstName }}?\n  </div>\n  <div fxLayout=\"row\" fxLayoutAlign=\"center\" class=\"mt-16\">\n    <mat-star-rating\n      [rating]=\"rating\"\n      starCount=\"5\"\n      color=\"accent\"\n      (ratingUpdated)=\"onRatingChanged($event)\">\n    </mat-star-rating>\n  </div>\n  <div class=\"mt-16\">\n    2.  Please comment on why you have given {{ provider?.firstName }} the rating above.\n  </div>\n  <div fxLayout=\"row\" class=\"mt-16\">\n    <mat-form-field fxFlex>\n      <textarea placeholder=\"Comments\" matInput [formControl]=\"commentFormControl\"></textarea>\n      <mat-error>\n        Required\n      </mat-error>\n    </mat-form-field>\n  </div>\n  <div fxLayout=\"row\" class=\"mt-16\">\n    <div fxLayout=\"row\" fxLayoutAlign=\"end\" *ngIf=\"loading\">\n      <mat-spinner [diameter]=\"25\"></mat-spinner>\n    </div>\n    <button mat-raised-button (click)=\"submitRating()\" [disabled]=\"loading\">Submit</button>\n  </div>\n</div>\n<div class=\"mt-16 p-16\">\n  <div class=\"mat-title\">\n    Reviews\n  </div>\n  <provider-rating-item *ngFor=\"let rating of ratings\" [customerRating]=\"rating\"></provider-rating-item>\n  <div fxLayout=\"row\" fxLayoutAlign=\"center center\" *ngIf=\"loadingReviews\">\n    <mat-spinner [diameter]=\"25\"></mat-spinner>\n    <div class=\"ml-8\">Loading...</div>\n  </div>\n</div>\n\n",
            styles: [""]
        })
    ], ProviderRatingWidgetComponent);
    return ProviderRatingWidgetComponent;
}());
export { ProviderRatingWidgetComponent };
