// providers/esm5/lib/providers/star-rating/star-rating.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter, ViewEncapsulation } from '@angular/core';
import { MatSnackBar } from '@angular/material/snack-bar';
var StarRatingComponent = /** @class */ (function () {
    function StarRatingComponent(snackBar) {
        this.snackBar = snackBar;
        this.ratingUpdated = new EventEmitter();
        this.snackBarDuration = 2000;
        this.ratingArr = [];
    }
    StarRatingComponent.prototype.ngOnInit = function () {
        for (var index = 0; index < this.starCount; index++) {
            this.ratingArr.push(index);
        }
    };
    StarRatingComponent.prototype.onClick = function (rating) {
        console.log(rating);
        this.ratingUpdated.emit(rating);
        return false;
    };
    StarRatingComponent.prototype.showRatingLabel = function (level) {
        var result = '';
        switch (level) {
            case 1:
                result = 'Poor';
                break;
            case 2:
                result = 'Not Good';
                break;
            case 3:
                result = 'Fine';
                break;
            case 4:
                result = 'Really Good';
                break;
            case 5:
                result = 'Excellent';
                break;
        }
        return result;
    };
    StarRatingComponent.prototype.showIcon = function (index) {
        if (this.rating >= index + 1) {
            return 'star';
        }
        else {
            return 'star_border';
        }
    };
    StarRatingComponent.ctorParameters = function () { return [
        { type: MatSnackBar }
    ]; };
    __decorate([
        Input('rating')
    ], StarRatingComponent.prototype, "rating", void 0);
    __decorate([
        Input('starCount')
    ], StarRatingComponent.prototype, "starCount", void 0);
    __decorate([
        Input('color')
    ], StarRatingComponent.prototype, "color", void 0);
    __decorate([
        Output()
    ], StarRatingComponent.prototype, "ratingUpdated", void 0);
    StarRatingComponent = __decorate([
        Component({
            selector: 'mat-star-rating',
            template: "<button mat-icon-button [color]=\"color\" *ngFor=\"let ratingId of ratingArr;index as i\" [id]=\"'star_'+i\" (click)=\"onClick(i+1)\" [matTooltip]=\"showRatingLabel(ratingId+1)\" matTooltipPosition=\"above\">\n  <mat-icon class=\"yellow-icon\">\n    {{showIcon(i)}}\n  </mat-icon>\n</button>\n<mat-error *ngIf=\"starCount == null || starCount == 0\">\n  Star count is <strong>required</strong> and cannot be zero\n</mat-error>\n",
            encapsulation: ViewEncapsulation.Emulated,
            styles: ["button{height:20px;width:25px;line-height:20px}.yellow-icon{color:#fdd835}.yellow-icon svg{fill:#fdd835}"]
        })
    ], StarRatingComponent);
    return StarRatingComponent;
}());
export { StarRatingComponent };
export var StarRatingColor;
(function (StarRatingColor) {
    StarRatingColor["primary"] = "primary";
    StarRatingColor["accent"] = "accent";
    StarRatingColor["warn"] = "warn";
})(StarRatingColor || (StarRatingColor = {}));
