// providers/esm2015/lib/providers/provider-select/provider-select.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
import { StreamActivity, StreamClientService } from '@brightpath/activity-stream';
import { CustomerService, ProvidersService } from '@brightpath/scheduling';
let ProviderSelectComponent = class ProviderSelectComponent {
    constructor(providerService, customerService, streamClient, router) {
        this.providerService = providerService;
        this.customerService = customerService;
        this.streamClient = streamClient;
        this.router = router;
        this.loading = false;
        this.providers = [];
    }
    ngOnInit() {
        if (this.customer === null || this.customer === undefined) {
            // load the customer
            this.customerService.loadCustomer(this.customerId).subscribe((customerList) => {
                if (customerList !== null && customerList.length > 0) {
                    this.customer = customerList[0];
                }
            });
        }
        this.loadProviders();
    }
    loadProviders() {
        this.providerService.getAllProviders().then((providers) => {
            console.log(providers);
            this.providers = providers;
        });
    }
    onProviderSelected(newProvider) {
        console.log('the user wants this guy: ', newProvider);
        if (this.customer.preferredProvider !== null && this.customer.preferredProvider !== undefined) {
            // this person is switching providers.  Create an activity and CC it to the old provider's notification stream
            this.streamClient.getToken('timeline', this.customer.id).then((response) => {
                const token = response.token;
                const activity = new StreamActivity();
                activity.verb = 'cancelprovider';
                activity.actor = `customers:${this.customer.id}`;
                activity.target = `providers:${this.customer.preferredProvider}`;
                activity.to = [`notification:${this.customer.preferredProvider}`];
                activity.object = 'none';
                console.log('about to create the following', activity);
                this.streamClient.addActivity(activity, 'timeline', this.customer.id, token);
            });
        }
        this.customer.preferredProvider = newProvider.id;
        // TODO: Need to signal that the preferred provider has changed
        // this.auth.saveAuthIdentityToLocalStorage();
        this.customerService.saveCustomer(this.customer)
            .subscribe((response) => {
            this.streamClient.getToken('timeline', this.customer.id).then((tokenresponse) => {
                console.log('got the ticket honkey dorey');
                const token = tokenresponse.token;
                const activity = new StreamActivity();
                activity.verb = 'preferredprovider';
                activity.actor = `customers:${this.customer.id}`;
                activity.target = `providers:${this.customer.preferredProvider}`;
                activity.to = [`notification:${this.customer.preferredProvider}`];
                activity.object = 'none';
                console.log('about to create the following', activity);
                this.streamClient.addActivity(activity, 'timeline', this.customer.id, token);
                // now follow the new provider
                this.streamClient.follow(this.customer.id, 'timeline', this.customer.preferredProvider, 'providers');
            });
            // TODO: This shouldn't navigate, but should instead emit an event
            this.router.navigateByUrl('home');
        });
    }
};
ProviderSelectComponent.ctorParameters = () => [
    { type: ProvidersService },
    { type: CustomerService },
    { type: StreamClientService },
    { type: Router }
];
__decorate([
    Input()
], ProviderSelectComponent.prototype, "customerId", void 0);
__decorate([
    Input()
], ProviderSelectComponent.prototype, "customer", void 0);
ProviderSelectComponent = __decorate([
    Component({
        selector: 'provider-select',
        template: "<div fxLayout=\"column\" class=\"page-layout blank p-24\" fxLayoutGap=\"16px\" fusePerfectScrollbar>\n  <div fxLayout=\"row\" fxLayoutWrap fxLayoutGap=\"16px\">\n    <provider-widget (onSelectedProvider)=\"onProviderSelected($event)\" fxFlex=\"25\" class=\"mb-16\" *ngFor=\"let provider of providers\" [provider]=\"provider\"></provider-widget>\n  </div>\n</div>\n",
        styles: [""]
    })
], ProviderSelectComponent);
export { ProviderSelectComponent };
