// providers/esm2015/lib/providers/provider-summary/provider-summary.component.js
import { __decorate } from "tslib";
import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
import { Provider } from '@brightpath/common';
import { ProvidersService } from '@brightpath/scheduling';
let ProviderSummaryComponent = class ProviderSummaryComponent {
    constructor(providerService) {
        this.providerService = providerService;
    }
    ngOnInit() {
        // console.log('ngOnInit and tplRef === ' + this.tplRef);
        if (this.provider === undefined || this.provider === null) {
            // we need to have a providerId or there's a problem here
            if (this.providerId === undefined || this.providerId === null) {
                this.provider = new Provider({
                    firstName: 'None',
                    lastName: ''
                });
            }
            else {
                this.providerService.getProviderById(this.providerId).then((result) => {
                    this.provider = new Provider(result);
                }).catch((error) => {
                    console.log('An error occurred', error);
                });
            }
        }
    }
};
ProviderSummaryComponent.ctorParameters = () => [
    { type: ProvidersService }
];
__decorate([
    Input()
], ProviderSummaryComponent.prototype, "providerId", void 0);
__decorate([
    Input()
], ProviderSummaryComponent.prototype, "provider", void 0);
__decorate([
    ContentChild(TemplateRef)
], ProviderSummaryComponent.prototype, "tplRef", void 0);
ProviderSummaryComponent = __decorate([
    Component({
        selector: 'provider-provider-summary',
        template: "<ng-template [customTemplate]=\"tplRef\">\n  <div fxLayout=\"row\" fxLayoutGap=\"16px\" *ngIf=\"provider\">\n    <div>\n      <img class=\"avatar huge\" [src]=\"provider.imageUrl\">\n    </div>\n    <div fxFlex fxLayoutAlign=\"start center\" class=\"mat-h1\">\n      {{ provider.firstName }} {{ provider.lastName }}\n    </div>\n  </div>\n</ng-template>\n",
        styles: [""]
    })
], ProviderSummaryComponent);
export { ProviderSummaryComponent };
