// providers/esm5/lib/providers/provider-rating/provider-rating-widget/provider-rating-item/provider-rating-item.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import * as moment from 'moment';
var ProviderRatingItemComponent = /** @class */ (function () {
    function ProviderRatingItemComponent() {
        this.moment = moment;
    }
    ProviderRatingItemComponent.prototype.ngOnInit = function () {
    };
    __decorate([
        Input()
    ], ProviderRatingItemComponent.prototype, "customerRating", void 0);
    ProviderRatingItemComponent = __decorate([
        Component({
            selector: 'provider-rating-item',
            template: "<div class=\"my-16\">\n  <blockquote fxLayout=\"column\">\n    <div class=\"mb-16\">\n      <mat-star-rating\n        [rating]=\"customerRating.rating\"\n        starCount=\"5\"\n        color=\"accent\">\n      </mat-star-rating>\n    </div>\n    <div>\n      {{ customerRating.comments }}\n    </div>\n    <div class=\"secondary-text mt-8\">\n      {{ moment().format('MMMM YYYY')}}\n    </div>\n  </blockquote>\n</div>\n",
            styles: [""]
        })
    ], ProviderRatingItemComponent);
    return ProviderRatingItemComponent;
}());
export { ProviderRatingItemComponent };
