// providers/esm5/lib/providers/provider-private-message/provider-private-message.component.js
import { __decorate } from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { UUID } from 'angular2-uuid';
import { MatSnackBar } from '@angular/material/snack-bar';
import { StreamActivity, StreamClientService } from '@brightpath/activity-stream';
var ProviderPrivateMessageComponent = /** @class */ (function () {
    function ProviderPrivateMessageComponent(streamClientService, snackbar) {
        this.streamClientService = streamClientService;
        this.snackbar = snackbar;
        this.close = new EventEmitter();
    }
    ProviderPrivateMessageComponent.prototype.onPrivateMessage = function (privateMessage) {
        var _this = this;
        privateMessage.actor = 'customers:' + this.customerId;
        privateMessage.threadId = UUID.UUID();
        privateMessage.to = ["notification:" + this.providerId];
        privateMessage.target = "inbox:" + this.providerId;
        this.streamClientService.getToken('inbox', this.providerId).then(function (response) {
            var token = response.token;
            _this.streamClientService.addActivity(privateMessage, 'inbox', _this.providerId, token).then(function (privateMessageResponse) {
                _this.snackbar.open('Sent Private Message', 'OK', {
                    verticalPosition: 'top',
                    duration: 2000
                });
            });
            _this.close.emit(true);
        });
    };
    ProviderPrivateMessageComponent.prototype.ngOnInit = function () {
    };
    ProviderPrivateMessageComponent.ctorParameters = function () { return [
        { type: StreamClientService },
        { type: MatSnackBar }
    ]; };
    __decorate([
        Input()
    ], ProviderPrivateMessageComponent.prototype, "customerId", void 0);
    __decorate([
        Input()
    ], ProviderPrivateMessageComponent.prototype, "providerId", void 0);
    __decorate([
        Output()
    ], ProviderPrivateMessageComponent.prototype, "close", void 0);
    ProviderPrivateMessageComponent = __decorate([
        Component({
            selector: 'provider-private-message',
            template: "<div class=\"p-8\">\n  <div fxLayout=\"row\" fxLayoutAlign=\"center center\" class=\"message-box error\">\n    If you are in immediate need of counseling or support, please call 911.  Messages sent to your counselor can take up to 24 hours to be read and replied to\n  </div>\n  <div fxLayout=\"column\">\n    <div fxLayout=\"row\" class=\"mt-16\">\n      <lib-activity-editor (onActivity)=\"onPrivateMessage($event)\" fxFlex></lib-activity-editor>\n    </div>\n  </div>\n</div>\n",
            styles: [""]
        })
    ], ProviderPrivateMessageComponent);
    return ProviderPrivateMessageComponent;
}());
export { ProviderPrivateMessageComponent };
